web/packages/design/src/Modal/Modal.tsx

```diff
@@ -222,7 +232,18 @@ export default class Modal extends React.Component<ModalProps> {
         hiddenInDom={!open}
         modalCss={modalCss}
         data-testid="Modal"
-        ref={this.modalRef}
+        ref={el => {
+          this.modalEl = el;
+          const { modalRef } = this.props;
+          if (modalRef) {
```

Comment:
This works, but another way of solving this would be to keep the original `modalRef` (maybe rename it to `privateModalRef` or whatever) and then use [`mergeRefs`](https://github.com/gravitational/teleport/blob/28279490f114c3ee846b89988ba84ed26efa159f/web/packages/shared/libs/mergeRefs.ts) to pass two refs to the same element.

I don't think it's worth changing now, given your availability and the fact that `mergeRefs` is in the shared package, so it'd need to be moved to design first.

Reply:
Oh, wow, this is exactly the utility I've been looking for. But yeah, moving it to `design` (with 2 backports) is not something I'd like to do right now. :/